integr8ly/integreatly-operator
# MGDAPI-5275 - fix: disable cluster logging

# Issue link

https://issues.redhat.com/browse/MGDAPI-5275

# What

* Disable installation of cluster logging operator
* Cleanup installed cluster logging resources from previous installs
  * Logic is based from https://github.com/redhat-developer/observability-operator/blob/main/controllers/reconcilers/logging_installation/logging_installation_reconciler.go#L47-L83

# Verification steps



* Checkout master
* Install RHOAM
```
export LOCAL=false                                                   
make cluster/prepare/local
make code/run
```
* Once installation completes, stop the operator
* Checkout this branch
* Watch Cluster logging resources
```
watch "echo "CSV:"; oc get csv -n openshift-logging; echo "Subscription:"; oc get subscriptions -n openshift-logging; echo "Pods:"; oc get pods -n openshift-logging; echo "ClusterLogForwarders:"; oc get clusterlogforwarders -n openshift-logging; echo "ClusterLoggings:"; oc get clusterloggings -n openshift-logging;"
```
* Re-start Operator
```
export LOCAL=false
make code/rerun 
```
* Verify:
  * Cluster Logging operator subscription is removed
  * Cluster Logging operator csv is removed
  * Cluster Logging pod is removed
  * ClusterLogging CR is removed
  * ClusterLogForwarder CR is removed
